apm-agent-plugins/apm-es-restclient-plugin/apm-es-java-client-test/src/test/java/co/elastic/apm/esjavaclient/AbstractElasticsearchJavaTest.java

```diff
@@ -0,0 +1,525 @@
+import co.elastic.clients.elasticsearch.core.msearch.RequestItem;
+import co.elastic.clients.elasticsearch.core.msearch_template.TemplateConfig;
+import co.elastic.clients.elasticsearch.core.search.Hit;
+import co.elastic.clients.elasticsearch.core.search.HitsMetadata;
+import co.elastic.clients.elasticsearch.indices.CreateIndexRequest;
+import co.elastic.clients.elasticsearch.indices.CreateIndexResponse;
+import co.elastic.clients.elasticsearch.indices.DeleteIndexRequest;
+import co.elastic.clients.elasticsearch.indices.DeleteIndexResponse;
+import co.elastic.clients.elasticsearch.rollup.RollupSearchRequest;
+import co.elastic.clients.elasticsearch.rollup.RollupSearchResponse;
+import co.elastic.clients.json.JsonData;
+import jakarta.json.Json;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class AbstractElasticsearchJavaTest extends AbstractEsClientInstrumentationTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractElasticsearchJavaTest.class);
+
+    protected static final String USER_NAME = "elastic-user";
+    protected static final String PASSWORD = "elastic-pass";
+
+    protected static ElasticsearchClient client;
+    protected static ElasticsearchAsyncClient asyncClient;
+
+    @Test
+    public void testCreateAndDeleteIndex() throws IOException, ExecutionException, InterruptedException {
+        // Create an Index
+        doCreateIndex(new CreateIndexRequest.Builder().index(SECOND_INDEX).build());
+        validateSpanContentAfterIndexCreateRequest();
+
+        reporter.reset();
+
+        // Delete the index
+        doDeleteIndex(new DeleteIndexRequest.Builder().index(SECOND_INDEX).build());
+        validateSpanContentAfterIndexDeleteRequest();
+    }
+
+    @Test
+    public void testTryToDeleteNonExistingIndex() {
+        Throwable cause = null;
+        try {
+            doDeleteIndex(new DeleteIndexRequest.Builder().index(SECOND_INDEX).build());
+        } catch (ExecutionException ee) {
+            // async scenario
+            cause = ee.getCause();
+        } catch (Exception e) {
+            LOGGER.error("Exception during deleting non-existing index", e);
+            // sync scenario
+            cause = e;
+        }
+
+        assertThat(cause).isInstanceOf(ElasticsearchException.class);
+
+        ElasticsearchException ee = (ElasticsearchException) cause;
+        assertThat(ee.status()).isEqualTo(404);
+
+        assertThat(cause).isNotNull();
+
+//        no errors captured, because in this case event come to ResponseListener#onSuccess
```

Comment:
This is weird..
I'll check if this is not a client bug. 
In the meantime I'll leave commented out and extend span validation

Reply:
This is the response I got from the Java client team: Java API client explicitly ask the REST client to consider status codes 400,401,403,404,405 as non-error. They are handled at a higher level and may or may not lead to an exception to be thrown depending on the endpoint that was called.
Since we instrument the low level client, we can't capture such errors as the `onSuccess` will be invoked rather than `onFailure`. In order to capture that, we would need to add a different instrumentation to the Java API client, but this is out of scope for now and not necessarily something we want to do.